packages/opentelemetry-shim-opentracing/src/shim.ts

```diff
@@ -163,60 +166,70 @@ export class TracerShim extends opentracing.Tracer {
     const oTelSpanBaggage: api.Baggage = spanContextShim.getBaggage();
 
     if (!carrier || typeof carrier !== 'object') return;
-    switch (format) {
-      case opentracing.FORMAT_HTTP_HEADERS:
-      case opentracing.FORMAT_TEXT_MAP: {
-        api.propagation.inject(
-          api.setBaggage(
-            api.setSpanContext(api.ROOT_CONTEXT, oTelSpanContext),
-            oTelSpanBaggage
-          ),
-          carrier
-        );
-        return;
-      }
-      case opentracing.FORMAT_BINARY: {
-        api.diag.warn(
-          'OpentracingShim.inject() does not support FORMAT_BINARY'
-        );
-        // @todo: Implement binary formats
-        return;
-      }
-      default:
+
+    if (format === opentracing.FORMAT_BINARY) {
+      api.diag.warn('OpentracingShim.inject() does not support FORMAT_BINARY');
+      // @todo: Implement binary format
+      return;
+    }
+
+    const propagator = this._getPropagator(format);
+    if (propagator !== undefined) {
+      const context = api.setBaggage(
+        api.setSpanContext(api.ROOT_CONTEXT, oTelSpanContext),
+        oTelSpanBaggage
+      );
+      propagator.inject(context, carrier, api.defaultTextMapSetter);
     }
   }
 
   _extract(format: string, carrier: unknown): opentracing.SpanContext | null {
+    if (format === opentracing.FORMAT_BINARY) {
+      api.diag.warn('OpentracingShim.extract() does not support FORMAT_BINARY');
+      // @todo: Implement binary format
+      return null;
+    }
+
+    const propagator = this._getPropagator(format);
+    if (propagator !== undefined) {
+      const context: api.Context = propagator.extract(
+        api.ROOT_CONTEXT,
+        carrier,
+        api.defaultTextMapGetter
+      );
+      const spanContext = api.getSpanContext(context);
+      const baggage = api.getBaggage(context);
+
+      if (!spanContext) {
+        return null;
+      }
+      return new SpanContextShim(spanContext, baggage || createBaggage());
+    }
+    return null;
+  }
+
+  private _getPropagator(format: string): TextMapPropagator | undefined {
     switch (format) {
+      case opentracing.FORMAT_TEXT_MAP:
+        if (this._propagators?.textMapPropagator !== undefined) {
```

Comment:
I think this could be written in a single line: `return this._propagators?.textMapPropagator ?? api.propagation`

Reply:
addressed in the new commit